src/util/Logger.ts

```diff
@@ -0,0 +1,89 @@
+import * as vscode from 'vscode';
+
+export type LogLevel = 'error' | 'warn' | 'info' | 'debug' | 'debug2';
+
+export type LogLevelHandler = (label: string, ...args: unknown[]) => void;
+
+export type LogHandler = Record<LogLevel, LogLevelHandler>;
+
+/**
+ * Simple logger delegate that can be used to log messages to a set of handlers.
+ * Messages will include a label for scoping log messages.
+ *
+ * Handlers can be statically registered via:
+ *
+ *   Logger.handlers.add(handler);
+ *
+ * Then modules can create labeled loggers and use them to log messages:
+ *
+ *   const logger = new Logger('MyModule');
+ *   logger.info('Hello, world!');
+ */
+export class Logger {
+  static handlers: Set<LogHandler> = new Set();
+
+  /**
+   * Register log handler that logs to console.
+   */
+  static addConsoleHandler = () => {
+    Logger.handlers.add({
+      /* eslint-disable no-console */
+      error: (label, ...args) => console.error(`[${label}] ERROR:`, ...args),
+      warn: (label, ...args) => console.warn(`[${label}] WARN:`, ...args),
+      info: (label, ...args) => console.info(`[${label}] INFO:`, ...args),
+      debug: (label, ...args) => console.debug(`[${label}] DEBUG:`, ...args),
+      debug2: (label, ...args) => console.debug(`[${label}] DEBUG2:`, ...args),
```

Comment:
You should set these to `console` methods with `bind` instead of redirecting. Otherwise, each log line printed will point to these lines instead of where the logger is being called from. See how we do it in the web-client-ui Logger.ts: https://github.com/deephaven/web-client-ui/blob/0da547d3d31f49f4302b1c2034660871ead14476/packages/log/src/Logger.ts#L5

(Note: I didn't install/try this out so unsure if the line numbers even get printed out)

Reply:
fixed